features/fields_extensie.feature

```diff
@@ -1,126 +1,544 @@
+      | naam                      |
+      | voornamen                 | Johannnes Geurt |
+      | geslachtsnaam             | Janssen         |
+      | aanduidingNaamgebruik     | E               |
+      En de persoon heeft de volgende geboorte gegevens
+      | naam                                      | waarde   |
+      | datum.type                                | Datum    |
+      | datum.datum (01.03.10)                    | 19860401 |
+      | land.code                                 | 6030     |
+      | plaats.code                               | 0518     |
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                                             |
+      | type                | RaadpleegMetBurgerservicenummer                    |
+      | burgerservicenummer | 999994086                                          |
+      | fields              | burgerservicenummer,geboorte.datum,geboorte.plaats |
+      Dan bevat de response de volgende gegevens
+      | naam                 | waarde       |
+      | burgerservicenummer  | 999994086    |
+      | geboorte.plaats.code | 0518         |
+      En bevat de response geen andere properties
+
+
+    Scenario: Fields bevat attributen die bij de geraadpleegde persoon geen waarde hebben
+      Gegeven het systeem heeft een persoon met de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999992934 |
+      | geslachtsaanduiding | M         |
+      En de persoon heeft de volgende naam gegevens
+      | naam                      |
+      | voornamen                 | Franklin |
+      | geslachtsnaam             | Groenen  |
+      | aanduidingNaamgebruik     | E               |
+      En de persoon heeft de volgende geboorte gegevens
+      | naam                                      | waarde   |
+      | datum.type                                | Datum    |
+      | datum.datum (01.03.10)                    | 19860401 |
+      | land.code                                 |          |
+      | plaats.code                               | 0518     |
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                                              |
+      | type                | RaadpleegMetBurgerservicenummer                     |
+      | burgerservicenummer | 999992934                                           |
+      | fields              | burgerservicenummer,geboorte.plaats,geboorte.land   |
+      Dan bevat de response de volgende gegevens
+      | naam                 | waarde       |
+      | burgerservicenummer  | 999992934    |
+      | geboorte.plaats      | 0518         |
+      En bevat de response geen andere properties
+
+    Scenario: gebruik fields in een collectie
+## Dit scenario licht niet het gebruik van Fields toe, maar de structuur van een andtwoord met een collectie. Die hoort hier volgens mij niet thuis.
```

Comment:
De bedoeling is om duidelijk te maken dat de envelop ("personen", voorheen "_embedded.ingeschrevenpersonen") niet in fields opgenomen wordt.

Wanneer je bijvoorbeeld personen zoekt met fields=personen.naam krijgt je een foutmelding met "reason": "Deel van de parameterwaarde niet correct: personen.naam."

Reply:
Ik heb nu een scenario toegevoegd met meerder responses, maar zoals je zelf al aangeeft gaat nu elk scenario over het retourneren van een collectie. 

Is het niet gebruiken van de "Personen" envelopnaam in fields dan niet een aparte rule ? 
En voegt die Rule egrbuiks gemak toe ? Waarom zou je dit verbieden en hier een foutmelding op geven. We zeggen overal dat je een unike naam, het hele pad of een deel van het pad mag toepassen in fields. Ik zie niet zo goed in waarom we dan op Personen.naam een foutmelding zouden moeten retourneren. Je weet namelijk als provider exact wat er gevraagd wordt.